transport/network/src/network.rs

```diff
@@ -229,10 +244,13 @@ where
                 entry.last_seen = current_time();
                 entry.last_seen_latency = latency;
                 entry.heartbeats_succeeded += 1;
+                // reset backoff in case of a successful ping
                 entry.backoff = self.cfg.backoff_min;
                 entry.update_quality(1.0_f64.min(entry.get_quality() + self.cfg.quality_step));
             } else {
-                entry.backoff = self.cfg.backoff_max.max(entry.backoff.powf(self.cfg.backoff_exponent));
+                // increase backoff in case of a failed ping, but cap it at the max backoff to
+                // prevent entries from being shut out
+                entry.backoff = self.cfg.backoff_max.min(entry.backoff.powf(self.cfg.backoff_exponent));
```

Comment:
Nice catch, validator should make sure though that the backoff_max cannot be a negative value.

Reply:
ok, done